### PR TITLE
Convert to Rails Engine

### PR DESCRIPTION
**Pull request by mejackreed**
This PR does the following:
- converts gem to a Rails Engine
- Adds in standardized date formats 
- Recompiles the fonts in new directory structure (Rails Engine)
- Adds engine_cart for dev and testing
- Creates `StyleGuide` routes and controllers to test application and for consumers for reference for gem consumers
  - Note routes only get added when not in production
- Adds specs for colors and icons. This way we can make expectations on things that should be present. Hopefully this can help with maintainability for long term.
- Adds travis support

### CODE REVIEW

File: lib/sul_styles/engine.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module SulStyles
+  class Engine < ::Rails::Engine
+    isolate_namespace SulStyles
+
+    config.after_initialize do |app|
+      unless Rails.env.production?
```
Comment (cbeer) [inline lib/sul_styles/engine.rb:12]: What's this about?

Comment (jkeck) [inline lib/sul_styles/engine.rb:12]: @cbeer this is something that @mejackreed and I were talking about the other day, which is that SULStyles could provide a style guide page that devs could look at as a palette for development and SUL Styles could use for to do automated testing against. I think we assumed that this style-guide page shouldn't necessarily be available in your app when running in production mode (similar to the teaspoon testing page).

Comment (cbeer) [inline lib/sul_styles/engine.rb:12]: Ok. I don't like mounting it at `/`, though. Why not mount it at `/styleguides` (or whatever) and have the engine set the root route?

Comment (mejackreed) [inline lib/sul_styles/engine.rb:12]: Updated to mount at `/styleguides`

**Comment by mejackreed** [comment]
Also the styleguide pages you get back look like:

## Colors

screen shot 2015-07-31 at 11 34 03 am

## Icons

screen shot 2015-07-31 at 11 34 21 am

**Pull request merged**